Various additions to documentation (#703)
cursorless-dev/cursorless: docs/user/README.md

For programming languages where Cursorless has rich parse tree support, we support modifiers that expand to the nearest containing function, class, etc. See [the source code](../../src/languages/constants.ts) for a list of supported languages. Below is a list of supported scope types, keeping in mind that this table can sometimes lag behind the actual list. Your cheatsheet (say "cursorless help") will have the most up-to-date list.

| Term                | Syntactic element                                                 |
| ------------------- | ----------------------------------------------------------------- |

> | **Language Scopes** | Scopes that rely a language's syntax                              |

Reviewer: Per other comment, I'd remove this line

##### `"file"`

The word '`"file"` can be used to expand the target to refer to the entire file.

- `"copy file"`
- `"take file"`
- `"take file blue air"`

For example, `"take file [blue] air"` selects the file including the token containing letter 'a' with a blue hat.

##### `"head & tail"`

The words `"head"` and `"tail"` can be used to create a selection from the input mark, through the beginning or end of the line, respectively.

- `"take head"`: select from the cursor the start of the line
- `"take tail"`: select from the cursor to the end of the line
- `"take head air"`: selects the mark through to start of the line
- `"take tail air"`: selects the mark through to the end of the line

##### `"block"`

> The `"block"` modifier expands to above and below the target to select lines until an empty line is reached.

Reviewer: ```suggestion
The `"block"` modifier expands above and below the target to select lines until an empty line is reached.
```

##### `"file"`

The word '`"file"` can be used to expand the target to refer to the entire file.

- `"copy file"`
- `"take file"`
- `"take file blue air"`

For example, `"take file [blue] air"` selects the file including the token containing letter 'a' with a blue hat.

##### `"head & tail"`

The words `"head"` and `"tail"` can be used to create a selection from the cursor's position, through the beginning or end of the line, respectively.

- `"take head"`: select from the cursor the start of the line
- `"take tail"`: select from the cursor to the end of the line
- `"take head spun"`
- `"take tail spun"`

> When you have a single cursor point and issue the first two commands, a selection will be made from that position to either the beginning or end of the file. If you have any code highlighted or address a mark(see last two examples) a selection joining the existing selection plus either the beginning or end of the line.

Reviewer: Not sure these lines help, and I think are a bit confusing.  Prob better to just add a note to the "spun" lines above

##### `"paragraph"`

> The `"paragraph"` modifier expands to above and below the target to select lines until an empty line is reached.

Reviewer: 👍

- `"take paragraph"`
- `"take paragraph <TARGET>"`

##### `"paint"`

> The `"paint"` modifier is useful when selecting trailing delimiters or multiple nodes. To be clear, Cursorless tries to do the correct thing selecting a scope or when deleting `"value"` or `"key"`. However, sometimes paint is useful.

Reviewer: I'd put this note at the end, as it doesn't say what "paint" actually does.  Can lead with the line below the examples

- `"take paint"`

> - `"take paint <TARGET>"`

Reviewer: would prob be good to add an example with some example text to show what it does

> Both of the commands will expand from a mark and return a selection up until a leading and trailing whitespace character.

Reviewer: ```suggestion
Both of the commands will expand from the mark forward and backward to include all adjacent non-whitespace characters
```

`move blue air to green bat`
Replaces the token containing letter 'b' with a green hat using the token containing letter 'a' with a blue hat, and the delete the latter token.

### Slice

The slice command is used to add multiple cursors to the editor. Each cursor is inserted at the same column on each row requested within the command.

> - `"pre <TARGET 1> slice <TARGET 2>"`
> - `"pre <TARGET 1> slice past <TARGET 2>"`: Add cursors from the first target through to the second target's line(inclusive end)

Reviewer: ```suggestion
- `"pre <TARGET 1> slice past <TARGET 2>"`: Add cursors from the first target through to the second target's line(inclusive end)
- `"pre <TARGET 1> slice <TARGET 2>"`: Shortened version of above `"slice past"` command
```

- `"pre <TARGET 1> slice until <TARGET 2>"`: Add cursors until the second target's line(non-inclusive end)
- `"pre <TARGET 1> slice between <TARGET 2>"`: Add cursors between first and second target's lines(non-inclusive start and end)

> 

Reviewer: Prob worth adding a couple concrete examples, eg `"pre air slice bat"`, `"pre air slice end of block"`

- `"round wrap <TARGET>"`: wraps the target with parentheses
- `"box wrap <TARGET>"`: wraps the target with square brackets

### Slice

> The slice command is used to add multiple cursors to the editor. Each cursor is inserted at the same column on each row requested within the command.

Reviewer: fwiw you can use with things other than "pre", eg "take", "chuck tail air slice bat", etc

- `"pre <TARGET 1> slice <TARGET 2>"`

> - `"pre <TARGET 1> slice (past|until) <TARGET 2>"`

Author: Great, thanks.

The first command is equivalent to the second command when `"past"` is used. When `"until"` is used in the second command, cursors are inserted from the first target through to the line above the second target.

### Reverse/Shuffle/Sort

> These commands are to be paired with the `"item"` scope and can be used to reorder items within a list.

Reviewer: You can really pair them with anything you want.  I will sometimes do eg "sort line air slice bat", or select a bunch of things and say "sort this", etc

> These commands are to be paired with the `"item"` scope and can be used to reorder items within a list.

Reviewer: You can really pair them with anything you want.  I will sometimes do eg "sort line air slice bat", or select a bunch of things and say "sort this", etc
